contrib/mypy/tests/python/pants_test/contrib/mypy/tasks/test_mypy.py

```diff
@@ -0,0 +1,50 @@
+from pants.contrib.mypy.tasks.mypy_task import MypyTask, MypyTaskError
+from pants_test.task_test_base import TaskTestBase
+from pants.backend.python.targets.python_library import PythonLibrary
+
+
+class MyPyTests(TaskTestBase):
+
+  @classmethod
+  def task_type(cls):
+    return MypyTask
+
+  def test_throws_no_warning_on_all_whitelisted_target_roots(self) -> None:
+    t1 = self.make_target('t1', PythonLibrary, tags=['type_checked'])
+    t2 = self.make_target('t2', PythonLibrary, tags=['type_checked'])
+    task = self.create_task(self.context(target_roots=[t1, t2]))
+    self.set_options(whitelist_tag_name='type_checked')
+    task.execute()
+
+  def test_throws_no_warning_on_some_whitelisted_target_roots_but_all_whitelisted_in_context(self) -> None:
+    t1 = self.make_target('t1', PythonLibrary)
+    t2 = self.make_target('t2', PythonLibrary, tags=['type_checked'])
+    t3 = self.make_target('t3', PythonLibrary, tags=['type_checked'], dependencies=[t2])
+    task = self.create_task(self.context(target_roots=[t1, t3]))
+    self.set_options(whitelist_tag_name='type_checked')
+    task.execute()
+
+  def test_throws_warning_on_some_whitelisted_target_roots_but_all_whitelisted_in_context(self) -> None:
+    t1 = self.make_target('t1', PythonLibrary)
+    t2 = self.make_target('t2', PythonLibrary, tags=['something_else'])
+    t3 = self.make_target('t3', PythonLibrary, tags=['type_checked'], dependencies=[t2])
+    task = self.create_task(self.context(target_roots=[t1, t3]))
+    self.set_options(whitelist_tag_name='type_checked')
+    task.execute()
+    self.assertTrue(task.WARNING_MESSAGE)
```

Comment:
1. asserting True on str is relatively anti-pattern.
2. I think a good alternative here regarding your comment 
> added this field (`WARNING_MESSAGE `) for testing purposes, didn't seem worth spending a lot of time trying to figure out a way to keep track of logging/warning messages, so I used this workaround for now.

is to mock out `_whitelist_warning`, then `assert_called()`. E.g. https://github.com/pantsbuild/pants/blob/94d06ad77779c7d514b81e63d37c26ea284c0cba/tests/python/pants_test/backend/jvm/tasks/test_coursier_resolve.py#L211-L221